test/plugins/test_smartplaylist.py

```diff
@@ -241,6 +241,58 @@ def test_playlist_update_output_m3u8(self):
             + b"http://beets:8337/files/tagada.mp3\n",
         )
 
+    def test_playlist_update_output_extm3u_fields(self):
+        spl = SmartPlaylistPlugin()
+
+        i = MagicMock()
+        type(i).artist = PropertyMock(return_value="Fake Artist")
+        type(i).title = PropertyMock(return_value="fake Title")
+        type(i).length = PropertyMock(return_value=300.123)
+        type(i).path = PropertyMock(return_value=b"/tagada.mp3")
+        a = {"id": 456, "genre": "Fake Genre"}
+        i.__getitem__.side_effect = a.__getitem__
+        i.evaluate_template.side_effect = lambda pl, _: pl.replace(
+            b"$title",
+            b"ta:ga:da",
+        ).decode()
+
+        lib = Mock()
+        lib.replacements = CHAR_REPLACE
+        lib.items.return_value = [i]
+        lib.albums.return_value = []
+
+        q = Mock()
+        a_q = Mock()
+        pl = b"$title-my<playlist>.m3u", (q, None), (a_q, None)
+        spl._matched_playlists = [pl]
+
+        dir = bytestring_path(mkdtemp())
+        config["smartplaylist"]["output"] = "extm3u"
+        config["smartplaylist"]["relative_to"] = False
+        config["smartplaylist"]["playlist_dir"] = py3_path(dir)
+        config["smartplaylist"]["fields"] = ["id", "genre"]
+        try:
+            spl.update_playlists(lib)
+        except Exception:
+            rmtree(syspath(dir))
+            raise
+
+        lib.items.assert_called_once_with(q, None)
+        lib.albums.assert_called_once_with(a_q, None)
+
+        m3u_filepath = path.join(dir, b"ta_ga_da-my_playlist_.m3u")
+        self.assertExists(m3u_filepath)
+        with open(syspath(m3u_filepath), "rb") as f:
+            content = f.read()
+        rmtree(syspath(dir))
+
+        self.assertEqual(
+            content,
+            b"#EXTM3U\n"
+            + b'#EXTINF:300 id="456" genre="Fake Genre",Fake Artist - fake Title\n'
```

Comment:
Something like this: Add a config example and the resulting line on the playlist file to the docs.